src/model/guild/guild_id.rs

```diff
@@ -203,9 +232,12 @@ impl GuildId {
     /// Also can return [`Error::Http`] if the current user lacks permission.
     ///
     /// [Ban Members]: Permissions::BAN_MEMBERS
-    #[inline]
-    pub async fn ban(self, http: impl AsRef<Http>, user: impl Into<UserId>, dmd: u8) -> Result<()> {
-        self._ban(http, user.into(), dmd, None).await
+    #[cache_check = if let Some(cache) = cache_http.cache() {
+        self.require_perms(&cache_http, Permissions::BAN_MEMBERS).await?;
+        self.check_hierarchy(cache, user)?;
+    }]
+    pub async fn ban(self, cache_http: impl CacheHttp, user: UserId, dmd: u8) -> Result<()> {
```

Comment:
A shame that `impl Into<UserId>` had to be changed to `UserId`. The cache check here is the reason for that change, right? Is there any way to possibly resolve this? If not, the signature change is causing a doctest to fail, so just change that callsite to pass `user.id` instead of `user`.

Reply:
Oh, are the `impl Into<UserId>` something we want to keep?

Reply:
Unsure, but quite a lot of other methods take `impl Into<UserId>`, including `kick` and `kick_with_reason`. The only ones that take a raw `UserId` are `audit_logs` and `_ban`.